### PR TITLE
Add: Recovery phone validation flow

### PR DESCRIPTION
**Pull request by southp**
## Summary
This PR adds the validation flow for account recovery phones. e.g.
demo-validate-phone

Note that it is only available for users without binding 2fa sms numbers, simply because we blocks those users from editing recovery phone numbers at the moment. After deploying this PR, I will open another PR for removing the block. Just wanted to make the whole process gradual to play safe.

## Test Plan
Please make sure you have deployed D3859 to your test environment. Otherwise, you won't receive the validation code. All the following steps assume you are logged in as a user without binding a 2FA sms number.

Case 1: Add a recovery phone number

* Make sure the recovery phone field is empty. If not, remove it.
* Enter a valid phone number, see the validation notice prompt, and then you should receive a validation code shortly.
* Enter the code, click validate, and see it successful.
* Refresh the page. There shouldn't be any validation notice, which means the number is validated.

Case 2: Update a recovery phone number

* Make sure the recovery phone field is not empty.
* Update the number, see the validation notice prompt, and then you should receive a validation code shortly.
* Enter the code, click validate, and see it successful.
* Refresh the page. There shouldn't be any validation notice, which means the number is validated.

Case 3: Request a new validation code

* Add or update the phone field, so that the number is not validated.
* Refresh the page, and you should find a "Resend" button underneath the validation notice.
* Hit "Resend", and then you should receive a new validation code shortly.
* Use the code to validate the number in the same way as above.

### CODE REVIEW

**Comment by jordwest** [comment]
Works for me, I just had a few comments before going through the code:

 - Could we move the Resend button to the notice? At first I didn't realize that it was possible to resend, as I was looking for the button in the notice as with the email notice.

security_checkup_ _wordpress_com

 - The validation code text field could do with a label, it's not totally clear what its there for.

 - Might be worth tagging this for Design Review

**Comment by southp** [comment]
After a1d1a4efd401e88e7e10d81c1bdc1e0d5406703d and 6aa3a246ad6e48761fb75de00d2aa8ac5584c9a4, it looks like this now:
image

The notice disappears after hitting the resend button, and shows after a refresh after updating the phone number. Just like how it works in the email validation.

File: client/state/account-recovery/settings/selectors.js [hunk 1]
```diff
@@ -46,9 +50,21 @@ export const hasSentAccountRecoveryEmailValidation = ( state ) => {
...
+};
+
 export const shouldPromptAccountRecoveryEmailValidationNotice = ( state ) => {
 	return ! isAccountRecoveryEmailActionInProgress( state ) &&
 		!! getAccountRecoveryEmail( state ) &&
 		! isAccountRecoveryEmailValidated( state ) &&
 		! hasSentAccountRecoveryEmailValidation( state );
 };
+
+// It doesn't include hasSentAccountRecoveryPhoneValidation() like shouldPromptAccountRecoveryEmailValidationNotice() intentionall.
```
Comment (jordwest) [inline client/state/account-recovery/settings/selectors.js]: intentionall**y** :)

File: client/me/security-checkup/recovery-phone-validation-notice.jsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+					autoComplete="off"
+					disabled={ false }
+					placeholder={ translate( 'e.g. 12345678' ) }
+					onChange={ this.onChange }
+					value={ this.state.candidateCode }
+				/>
+				<FormButtonsBar className="security-checkup__recovery-phone-validation-buttons">
+					<FormButton
+						isPrimary={ true }
+						disabled={ isValidating || validationCodeLength !== this.state.candidateCode.length }
```
Comment (southp) [inline client/me/security-checkup/recovery-phone-validation-notice.jsx]: True. Fixed by 896b4cd923c566c5504727e51efde7f65c5ca24b

**Comment by evilluendas** [comment]
I think it looks great, and works as expected 👍 It worries me a bit that the Resend button is not show initially (only after refresh), but on the other hand I don't think we'd like to have the users clicking it repeatedly and sending a hundred sms if their cell network is a bit slow.

Good job!

**Comment by southp** [comment]
@jordwest After discussing with @evilluendas, we have decided to leave the validation notices there to have a stronger prompt to encourage users to validate. Could you help me take a quick look if the newly-added code is fine? Thanks :)

**Pull request merged**